### PR TITLE
add-cni-csi-coredns

### PR DESCRIPTION
**Pull request by calvix**
towards https://github.com/giantswarm/roadmap/issues/2609

pretty similar to what this PR does https://github.com/giantswarm/cluster-aws/pull/294

the only difference is that we need to somehow inject a proper EKS API endpoint into cilium so here I was inspired by https://github.com/giantswarm/cluster-cloud-director/blob/main/helm/cluster-cloud-director/templates/update-values-hook-job.yaml

But instead of doing `helm post install` hook which seems pretty unpredictable I just create a job that will wait until `Cluster` CR is populated with `.spec.ControlPlaneEndpoint` and creates ConfigMap from that value which is used in the `HelmRelease` for cilium to inject the proper API URL,  this will not cause any issues during updates (which is not the case for the  helm post-install hook which is rerun every time there is an update)

### CODE REVIEW

**Review by bdehri — APPROVED** [review]

File: helm/cluster-eks/templates/update-values-job.yaml [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        runAsUser: {{ include "securityContext.runAsUser" $ }}
+        runAsGroup: {{ include "securityContext.runAsGroup" $ }}
+      containers:
+        - name: create-cp-cm
+          image: "{{ .Values.kubectlImage.registry }}/{{ .Values.kubectlImage.name }}:{{ .Values.kubectlImage.tag }}"
+          command:
+            - "/bin/sh"
+            - "-xc"
+            - |
+              while [ -z "${control_plane_endpoint}" ] ; do
```
Comment (AndiDog) [inline helm/cluster-eks/templates/update-values-job.yaml:70]: Probably we should set `activeDeadlineSeconds` since this could otherwise run forever

**Review by AndiDog — COMMENTED** [review]
LGTM

File: helm/cluster-eks/templates/update-values-job.yaml [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+                control_plane_endpoint=$(kubectl get cluster -n {{ $.Release.Namespace }} {{ include "resource.default.name" $ }} -o 'jsonpath={.spec.controlPlaneEndpoint.host}')
```
Comment (fiunchinho) [inline helm/cluster-eks/templates/update-values-job.yaml:71]: should we get the value from the `AWSManagedCluster.spec.controlPlaneEndpoint` field instead? To be honest, I'm not sure about the difference between the two.

Comment (calvix) [reply helm/cluster-eks/templates/update-values-job.yaml:71]: AFAIK the value in `Cluster` CR is universal and should be same across all providers so should be fine with just that

File: helm/cluster-eks/templates/update-values-job.yaml (hunk 1 above)
Comment (calvix) [reply helm/cluster-eks/templates/update-values-job.yaml:71]: done

**Pull request merged**